crates/red_knot_workspace/src/site_packages.rs

```diff
@@ -8,15 +8,67 @@
 //! reasonably ask us to type-check code assuming that the code runs
 //! on Linux.)
 
+use std::fmt;
 use std::io;
+use std::num::NonZeroUsize;
+use std::ops::Deref;
 
 use ruff_db::system::{System, SystemPath, SystemPathBuf};
 
-/// Attempt to retrieve the `site-packages` directory
-/// associated with a given Python installation.
+/// The value given by the `home` key in `pyvenv.cfg` files.
+///
+/// This is equivalent to `{sys_prefix_path}/bin`, and points
+/// to a directory in which a Python executable can be found.
+/// Confusingly, it is *not* the same as the [`PYTHONHOME`]
+/// environment variable that Python provides! However, it's
+/// consistent among all mainstream creators of Python virtual
+/// environments (the stdlib Python `venv` module, the third-party
+/// `virtualenv` library, and `uv`), was specified by
+/// [the original PEP adding the `venv` module],
+/// and it's one of the few fields that's read by the Python
+/// standard libary's `site.py` module.
+///
+/// Other values, such as the path to the Python executable or the
+/// base-executable `sys.prefix` value, are either only provided in
+/// `pyvenv.cfg` files by some virtual-environment creators,
+/// or are included under different keys depending on which
+/// virtual-environment creation tool you've used.
+///
+/// [`PYTHONHOME`]: https://docs.python.org/3/using/cmdline.html#envvar-PYTHONHOME
+/// [the original PEP adding the `venv` module]: https://peps.python.org/pep-0405/
+#[derive(Debug, PartialEq, Eq)]
+struct PythonHomePath(SystemPathBuf);
+
+impl PythonHomePath {
+    fn new(path: impl AsRef<SystemPath>, system: &dyn System) -> Option<Self> {
+        let path = path.as_ref();
+        let canonicalized = system
+            .canonicalize_path(path)
+            .unwrap_or_else(|_| path.to_path_buf());
```

Comment:
Is it important that we canonicalize the path here or do we need an absolute path? If it is a relative path, is it relative to the directory in which the CLI runs or the workspace?

Reply:
Yes, it's very important that we resolve symlinks here: see my extended conversation with @carljm on my prior PR: https://github.com/astral-sh/ruff/pull/12716#discussion_r1705853339. System installations created by e.g. homebrew seem to make heavy use of symlinks so that they put `site-packages` in odd places while still nominally adhering to Python's expected installation layout. I can add a comment